glide-core/src/client/value_conversion.rs

```diff
@@ -379,9 +369,99 @@ pub(crate) fn convert_to_expected_type(
+           3) "engine"
+           4) "LUA"
+           5) "functions"
+           6) 1) 1) "name"
+                 2) "myfunc1"
+                 3) "description"
+                 4) (nil)
+                 5) "flags"
+                 6) (empty array)
+              2) 1) "name"
+                 ...
+        2) 1) "library_name"
+           ...
+
+        RESP3 response
+        1) 1# "library_name" => "mylib1"
+           2# "engine" => "LUA"
+           3# "functions" =>
+              1) 1# "name" => "myfunc1"
+                 2# "description" => (nil)
+                 3# "flags" => (empty set)
+              2) 1# "name" => "myfunc2"
+                 ...
+        2) 1# "library_name" => "mylib2"
+           ...
+        */
+        ExpectedReturnType::ArrayOfMapsRecursive => match value {
+            // empty array, or it is already contains a map (RESP3 response) - no conversion needed
+            Value::Array(ref array) if array.is_empty() || matches!(array[0], Value::Map(_)) => {
+                Ok(value)
+            }
+            Value::Array(array) => {
+                convert_array_of_flat_maps(array, Some(ExpectedReturnType::ArrayOfMaps))
+            }
+            // cluster (multi-node) response - go recursive
+            Value::Map(map) => convert_map_entries(
+                map,
+                Some(ExpectedReturnType::BulkString),
+                Some(ExpectedReturnType::ArrayOfMapsRecursive),
+            ),
+            _ => Err((
+                ErrorKind::TypeError,
+                "Response couldn't be converted into a recursive array of maps",
+            )
+                .into()),
+        },
+        // Not used for a command, but used as a helper for `ArrayOfMapsRecursive` to process the inner map.
+        ExpectedReturnType::ArrayOfMaps => match value {
+            Value::Array(array) => {
+                convert_array_of_flat_maps(array, Some(ExpectedReturnType::StringOrSet))
+            }
+            // Due to recursion, it would be called to convert every map value, including simple strings, do nothing with them
+            Value::BulkString(_) | Value::SimpleString(_) | Value::VerbatimString { .. } => {
+                Ok(value)
+            }
+            _ => Err((ErrorKind::TypeError, "Response couldn't be converted").into()),
+        },
+        // Not used for a command, but used as a helper for `ArrayOfMaps` to process the inner map.
+        // It may contain a string (name, description) or set (flags), or nil (description).
+        // The set is stored as array. See comment for `ArrayOfMapsRecursive`.
+        ExpectedReturnType::StringOrSet => match value {
+            Value::Nil => Ok(value),
+            Value::Array(_) => convert_to_expected_type(value, Some(ExpectedReturnType::Set)),
+            _ => convert_to_expected_type(value, Some(ExpectedReturnType::BulkString)),
+        },
     }
 }
 
+/// Similar to [`convert_array_to_map`], but converts a map to a map.
```

Comment:
Kind of confusingly worded. I think this is a little more clear:
```suggestion
/// Similar to [`convert_array_to_map`], but converts an array of value pairs to a map.
```

Reply:
Input argument is always a map
https://github.com/amazon-contributing/redis-rs/blob/e28a97877fc353497b35a0cd033bf47cb199b47e/redis/src/types.rs#L325

Reply:
okay, but saying "map to a map" sounds like you're just running the identity function (which you are not). What does this actually do?

Reply:
It converts keys and values inside the map. So input is a map and output is a map too.

Reply:
okay, I feel like the comment could describe it as such. Otherwise, the function looks like it just takes a map and returns it, based on the description. That being said, if you really want to, you can leave the comment as is. I don't think it'll matter much in the long run for such a minor documentation nit.

Reply:
Updated, thank you.